src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py

```diff
@@ -1477,7 +1464,7 @@ def test_sql_db_security_mgmt(self, resource_group, resource_group_2,
         email_addresses_expected = ['test1@example.com','test2@example.com']
         email_account_admins = True
 
-        self.cmd('sql db threat-policy update -g {} -s {} -n {}'
+        self.cmd('sql db threat-policy update -g {} -s {} -n {} --security-alert-policy-name default'
```

Comment:
@Juliehzl 
I have a question about the newly added `security-alert-policy-name` as required parameter in track2 Python SDK: https://github.com/Azure/azure-sdk-for-python/blob/ede50054a5b78e1a69785221d7da19a0bb817278/sdk/sql/azure-mgmt-sql/azure/mgmt/sql/operations/_database_security_alert_policies_operations.py#L121

this required parameter `security-alert-policy-name` has only one default value "default": https://github.com/Azure/azure-sdk-for-python/blob/ede50054a5b78e1a69785221d7da19a0bb817278/sdk/sql/azure-mgmt-sql/azure/mgmt/sql/models/_sql_management_client_enums.py#L868 . Thus, in track1 python SDK, it was assigned internally: https://github.com/Azure/azure-sdk-for-python/blob/9d8c393590ee726bc0b0d02e8acafa523a79b694/sdk/sql/azure-mgmt-sql/azure/mgmt/sql/operations/_database_security_alert_policies_operations.py#L39

In the commands.py of this PR:
```
        g.generic_update_command('update',
                                 setter_name='create_or_update',
                                 custom_func_name='db_threat_detection_policy_update',
                                 )
```
it seems `generic_update_command` does not support assigning `security-alert-policy-name` without passing in it explicitly as `--security-alert-policy-name default` in the command line. But adding one more `--security-alert-policy-name default` as a mandatory parameter will be the breaking change of the old `sql db threat-policy update` command.

Do you know how can I assign the  `security-alert-policy-name` parameter in `generic_update_command`? 
Thanks!

Reply:
Then you need to rewrite create_or_update() and pass security-alert-policy-name in your create_or_update.

Reply:
do you have an example? 
do you mean I can override `setter_name`? 
I assume setter_name is the call path of Python SDK interface and custom_func_name is the method to assign for `parameters` signature in `create_or_update`.

Reply:
cc: @ayeletshpigelman 'cause I might make the breaking change of `sql db threat-policy update` CLI command

Reply:
@ericshape You can simply assign the security-alert-policy-name parameter in the `custom_func` which is `db_threat_detection_policy_update`

PS. You can find the intro for `custom_func_name` of `generic_update_command` [here](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#generic-update-commands)

Reply:
do you know how to update two fields: `security-alert-policy-name` and `parameters` ? 

```
    def generic_update_command(self, name, getter_name='get', getter_type=None,
                               setter_name='create_or_update', setter_type=None, setter_arg_name='parameters',
                               child_collection_prop_name=None, child_collection_key='name', child_arg_name='item_name',
                               custom_func_name=None, custom_func_type=None, **kwargs):
                               
```
                        
via the generic_update_command  impl, it seems the custom_func_name only change `setter_arg_name='parameters'` in default one value ONLY.

```
        if self.setter_arg_name == 'parameters':
            result = cached_put(self.cmd, setter, **setterargs)
        else:
            result = cached_put(self.cmd, setter, setterargs[self.setter_arg_name],
                                setter_arg_name=self.setter_arg_name, **setterargs)             
                               
```

Reply:
@ericshape, here is an example in `Network` https://github.com/Azure/azure-cli/blob/afec34798bf79439fd49b75a6d82d0ad714a1f05/src/azure-cli/azure/cli/command_modules/network/commands.py#L1418, where you define your own custom function as setter instead of call `begin_create_or_update` function in SDK directly. In your custom function, you call `begin_create_or_update` function in SDK where you can pass both `security-alert-policy-name` (with value `default`) and `parameters`.

![image](https://user-images.githubusercontent.com/16348853/126584163-e331c41e-4870-41ea-9675-b598d403de6c.png)

Reply:
address the comment